File: src/main/java/com/generator/invoice/InvoiceGeneratorApplication.java
package com.generator.invoice;

import com.generator.invoice.command.GenerateNewInvoiceCommand;
import org.springframework.boot.CommandLineRunner;
import org.springframework.boot.SpringApplication;
import org.springframework.boot.autoconfigure.SpringBootApplication;
import picocli.CommandLine;

@SpringBootApplication
public class InvoiceGeneratorApplication implements CommandLineRunner {
    private final GenerateNewInvoiceCommand generateNewInvoiceCommand;

    public InvoiceGeneratorApplication(GenerateNewInvoiceCommand generateNewInvoiceCommand) {
        this.generateNewInvoiceCommand = generateNewInvoiceCommand;
    }

    public static void main(String[] args) {
        SpringApplication.run(InvoiceGeneratorApplication.class, args);
    }

    @Override
    public void run(String... args) {
        new CommandLine(generateNewInvoiceCommand).execute(args);
    }
}

File: src/main/kotlin/com/generator/invoice/InvoiceGeneratorApplication.kt
package com.generator.invoice

import com.generator.invoice.command.GenerateNewInvoiceCommand
import com.generator.invoice.config.InvoiceGeneratorProperties
import org.springframework.boot.CommandLineRunner
import org.springframework.boot.autoconfigure.SpringBootApplication
import org.springframework.boot.context.properties.EnableConfigurationProperties
import org.springframework.boot.runApplication
import picocli.CommandLine

@SpringBootApplication
@EnableConfigurationProperties(InvoiceGeneratorProperties::class)
class InvoiceGeneratorApplication(
    private val generateNewInvoiceCommand: GenerateNewInvoiceCommand
) : CommandLineRunner {
    override fun run(vararg args: String?) {
        CommandLine(generateNewInvoiceCommand).execute(*args)
    }
}

fun main(args: Array<String>) {
    runApplication<InvoiceGeneratorApplication>(*args)
}
